public/javascripts/items.js

```diff
@@ -7,28 +7,97 @@ const fetchParams = {
   credentials: 'include'
 }
 
-const params = completed =>
-  Object.assign( {}, fetchParams, { body: JSON.stringify({ completed }) })
+const RETURN_KEY = 13
 
-$(document).ready( () => {
-  $( '.title' ).click( event => {
-    const element = $( event.target )
-    const id = element.data( 'id' )
-    const completed = ! element.data( 'completed' )
-
-    fetch( `/items/${id}/completed`, params( completed ))
-      .then( result => result.json() )
-      .then( json => {
-        if( json.success ) {
-          element.data( 'completed', completed )
-
-          const parent = $( `.item[data-id=${id}]` )
-          if( completed ) {
-            parent.addClass( 'completed' )
-          } else {
-            parent.removeClass( 'completed' )
-          }
+const params = data =>
+  Object.assign( {}, FETCH_PARAMS, { body: JSON.stringify( data ) } )
+
+const checkJsonForSuccessField = json => {
+  if( json.success ) {
+    Promise.resolve( json )
+  } else {
+    Promise.reject( json.message )
+  }
+}
+
+const selector = ( parent, id, tagToShow ) =>
+  `.${parent}[data-id=${id}] > ${tagToShow}`
+
+const toggle = ( elementToShow, elementToHide ) => {
+  $( elementToHide[ 0 ] ).addClass( 'hidden' )
+  $( elementToShow[ 0 ] ).removeClass( 'hidden' )
+}
+
+const clickToUpdate = parentClass => event => {
+  const elementToHide = $( event.target )
+  const id = elementToHide.data( 'id' )
+  const elementToShow = $( selector( parentClass, id, 'input' ) )
+
+  toggle( elementToShow, elementToHide )
+}
+
+const titleEdited = event => {
+  const elementToHide = $( event.target )
+  const id = elementToHide.data( 'id' )
+  const elementToShow = $( selector( 'title', id, 'span' ) )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedTitle = elementToHide[0].value
+    fetch( `/items/${id}`, params( { title: updatedTitle } ) )
+    .then( result => result.json() )
+    .then ( checkJsonForSuccessField )
+    .then( json => {
+      toggle( elementToShow, elementToHide )
+      $( elementToShow[0] ).html(updatedTitle)
+    })
+    .catch( error => console.log('error:', error) )
```

Comment:
This should not be here - it will prevent the server from sending a response.  We should remove the `catch` line and allow the default error handling to take effect so that the user will see an error message, at least.